server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/JMAPApiRoutes.scala

```diff
@@ -0,0 +1,55 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+package org.apache.james.jmap.routes
+
+import io.netty.handler.codec.http.HttpResponseStatus
+import org.apache.james.jmap.method.EchoMethod
+import reactor.core.scala.publisher.SMono
+import reactor.netty.http.server.{HttpServerRequest, HttpServerResponse}
+
+object JMAPApiRoutes {
+  private val ECHO_METHOD = new EchoMethod();
+  private val METHOD_NAME_PARAMETER = "method-name"
+}
+
+class JMAPApiRoutes {
+  private val echoMethod = JMAPApiRoutes.ECHO_METHOD
+  def post(httpServerRequest: HttpServerRequest, httpServerResponse: HttpServerResponse) = {
+    this.process(httpServerRequest, httpServerResponse)
+  }
+
+  def process(httpRequest: HttpServerRequest, httpServerResponse: HttpServerResponse) = {
+    httpRequest.param(JMAPApiRoutes.METHOD_NAME_PARAMETER) match {
```

Comment:
No that's not the way JMAP works.

There's a single API endpoint, request to it can ship several method calls

You can't base deserialization on such a easy trick.

Reply:
Can you explain it to me?